react/features/base/lib-jitsi-meet/functions.native.ts

```diff
@@ -1,31 +1,58 @@
 // @ts-ignore
-import { safeJsonParse } from '@jitsi/js-utils/json';
-import { NativeModules } from 'react-native';
+import { Worklets } from 'react-native-worklets-core';
 
 import { loadScript } from '../util/loadScript.native';
 
 import logger from './logger';
 
 export * from './functions.any';
 
-const { JavaScriptSandbox } = NativeModules;
-
 /**
  * Loads config.js from a specific remote server.
  *
  * @param {string} url - The URL to load.
  * @returns {Promise<Object>}
  */
 export async function loadConfig(url: string): Promise<Object> {
+    const configContext = Worklets.createContext('ConfigParser');
+
     try {
-        const configTxt = await loadScript(url, 10 * 1000 /* Timeout in ms */, true /* skipeval */);
-        const configJson = await JavaScriptSandbox.evaluate(`${configTxt}\nJSON.stringify(config);`);
-        const config = safeJsonParse(configJson);
+        const configTxt = await loadScript(url, 10 * 1000, true);
+
+        const parseConfigAsync = configContext.createRunAsync(function parseConfig(configText: string): string {
+            'worklet';
+            try {
+                // Used IIFE wrapper to capture config object from config.js
+                const configObj = eval(
+                    '(function(){\n'
+                    + configText
+                    + '\n; return (typeof config !== "undefined" ? config : globalThis.config); })()'
+                );
+
+                if (typeof configObj === 'undefined') {
+                    return 'Worklet_Error: config is undefined after eval()';
+                }
+
+                if (typeof configObj !== 'object' || configObj === null) {
+                    return 'Worklet_Error: config is not an object';
+                }
 
-        if (typeof config !== 'object') {
-            throw new Error('config is not an object');
+                return JSON.stringify(configObj);
+            } catch (err) {
+                return 'Worklet_Error:' + ((err as Error)?.message ?? String(err));
+            }
+        });
+
+        const workletConfig = await parseConfigAsync(configTxt);
+
+        if (workletConfig.startsWith('Worklet_Error:')) {
```

Comment:
This can't possibly work when workletConfig is an object because it doesn't have a `startsWith` function.